editor/plugins/particle_process_material_editor_plugin.cpp

```diff
@@ -0,0 +1,436 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "particle_process_material_editor_plugin.h"
+
+#include "editor/editor_property_name_processor.h"
+#include "editor/editor_settings.h"
+#include "editor/gui/editor_spin_slider.h"
+#include "scene/gui/base_button.h"
+#include "scene/gui/label.h"
+#include "scene/resources/particle_process_material.h"
+#include "scene/gui/box_container.h"
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_update_sizing() {
+	edit_size = range_edit_widget->get_size();
+	margin = Vector2(range_slider_left_icon->get_width(), (edit_size.y - range_slider_left_icon->get_height()) * 0.5);
+	usable_area = edit_size - margin * 2;
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_range_edit_draw() {
+	ERR_FAIL_COND(range_slider_left_icon.is_null());
+	ERR_FAIL_COND(range_slider_right_icon.is_null());
+	_update_sizing();
+
+	range_edit_widget->draw_rect(Rect2(margin, usable_area), Color(0.3, 0.3, 0.3, 1), false);
```

Comment:
Use width 1 for this rect, or min(round(EDSCALE), 1) if you think it should become wider at high editor scales. This way it won't have a missing corner.